api/src/opentrons/protocol_engine/state/frustum_helpers.py

```
@@ -286,3 +292,191 @@ def get_well_volumetric_capacity(
             )
             well_volume.append((next_f["topHeight"], bounded_volume))
     return well_volume
+
+
+def height_at_volume_within_section(
+    top_cross_section: Union[CircularBoundedSection, RectangularBoundedSection],
+    bottom_cross_section: Union[CircularBoundedSection, RectangularBoundedSection],
+    target_volume_relative: float,
+    frustum_height: float,
+) -> float:
+    """Calculate a height within a bounded section according to geometry."""
+    if top_cross_section["shape"] == bottom_cross_section["shape"] == "circular":
+        frustum_height = height_from_volume_circular(
+            volume=target_volume_relative,
+            top_radius=(top_cross_section["diameter"] / 2),
+            bottom_radius=(bottom_cross_section["diameter"] / 2),
+            total_frustum_height=frustum_height,
+        )
+    elif top_cross_section["shape"] == bottom_cross_section["shape"] == "rectangular":
+        frustum_height = height_from_volume_rectangular(
+            volume=target_volume_relative,
+            total_frustum_height=frustum_height,
+            bottom_width=bottom_cross_section["xDimension"],
+            bottom_length=bottom_cross_section["yDimension"],
+            top_width=top_cross_section["xDimension"],
+            top_length=top_cross_section["yDimension"],
+        )
+    return frustum_height
```

Raise not implemented here instead of a dummy value